metrics/web/lib/features/auth/presentation/widgets/auth_form.dart

```diff
@@ -0,0 +1,111 @@
+import 'package:email_validator/email_validator.dart';
+import 'package:flutter/material.dart';
+import 'package:metrics/features/auth/presentation/state/user_store.dart';
+import 'package:metrics/features/auth/presentation/strings/login_strings.dart';
+import 'package:metrics/features/auth/presentation/widgets/auth_input_field.dart';
+import 'package:states_rebuilder/states_rebuilder.dart';
+
+/// Shows an authentication form to sign in.
+class AuthForm extends StatefulWidget {
+  @override
+  _AuthFormState createState() => _AuthFormState();
+}
+
+/// The logic and internal state for the [AuthForm] widget.
+class _AuthFormState extends State<AuthForm> {
+  /// Global key that uniquely identifies the [Form] widget and allows validation of the form.
+  final _formKey = GlobalKey<FormState>();
+
+  /// Controls the email text being edited.
+  final TextEditingController _emailController = TextEditingController();
+
+  /// Controls the password text being edited.
+  final TextEditingController _passwordController = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    return Form(
+      key: _formKey,
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: <Widget>[
+          AuthInputField(
+            key: Key(LoginStrings.email),
+            label: LoginStrings.email,
+            controller: _emailController,
+            keyboardType: TextInputType.emailAddress,
+            validator: _validateEmail,
+          ),
+          AuthInputField(
+            key: Key(LoginStrings.password),
+            label: LoginStrings.password,
+            controller: _passwordController,
+            obscureText: true,
+            validator: _validatePassword,
+          ),
+          StateBuilder(
+            models: [Injector.getAsReactive<UserStore>()],
+            builder: (_, ReactiveModel<UserStore> store) {
+              return Padding(
+                padding: const EdgeInsets.symmetric(vertical: 10.0),
+                child: store.state.authErrorMessage.isNotEmpty
+                    ? Text(
+                        store.state.authErrorMessage,
+                        style: const TextStyle(color: Colors.red),
+                      )
+                    : null,
+              );
+            },
```

Comment:
with this, we need to add an extra margin to the submit button if the error message is empty, otherwise, the button will be too close to the password's input.